Compilation error for RTTTL component (2021.10.0.dev0)

### The problem

Compilation of rtttl component causes error in rtttl.cpp,
millis and delay are not declared in this scope.
(2021.10.0.dev0)

### Which version of ESPHome has the issue?

2021.10.0.dev0

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

2021.9.6

### What platform are you using?

ESP8266

### Board

esp01_1m

### Component causing the issue

rtttl

### Example YAML snippet

```yaml
# Substitutions
substitutions:
  # Device Names
  device_name: "powermeter111"
  friendly_name: "SDM"
  sub_name: "Dhouse"
  # Icon
  # https://materialdesignicons.com/
  # https://www.esphome.app/docs/icons
  main_icon: "mdi:gauge"

# CORE CONFIG
# https://esphome.io/components/esphome.html
esphome:
  name: "${device_name}"
  platform: ESP8266
  board: esp01_1m

  on_boot:
    priority: -10
    then:
      - rtttl.play: 'two_short:d=4,o=5,b=100:16e6,16e6'

# WIFI
# https://esphome.io/components/wifi.html
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  power_save_mode: none
  reboot_timeout: !secret reboot_timeout_wifi
  fast_connect: !secret fast_connect_wifi
  manual_ip:
    static_ip: !secret powermeter111
    gateway: !secret gateway
    subnet: !secret subnet

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "${friendly_name} Hotspot"
    password: !secret wifi_password

captive_portal:

# Enable logging
# https://esphome.io/components/logger.html
logger:
  # level: DEBUG
  level: INFO
  # logs:
  #  i2c.component: DEBUG
  # baud_rate: 115200
  baud_rate: 0  #disable uart log
  # esp8266_store_log_strings_in_flash: false

uart:
  rx_pin: GPIO3
  tx_pin: GPIO1
  baud_rate: 9600
  stop_bits: 1

# Enable Home Assistant API
# https://esphome.io/components/api.html
api:
  reboot_timeout: !secret reboot_timeout_api
  services:
    - service: powermeter_play
      variables:
        song_str: string
      then:
        - rtttl.play:
            rtttl: !lambda 'return song_str;'

# OTA
# https://esphome.io/components/ota.html
ota:

# WEBSERVER
# https://esphome.io/components/web_server.html
web_server:

# TIME
# https://esphome.io/components/time.html
time:
  - platform: homeassistant
    id: hass_time
    timezone: Europe/Warsaw

# ============================================================================
# ============================================================================
# ============================================================================

# https://esphome.io/components/binary_sensor
binary_sensor:
  - <<: !include common/binary_sensor/connection_status.config.yaml

# ============================================================================

# https://esphome.io/components/output
output:
  # https://esphome.io/components/rtttl.html
  - platform: esp8266_pwm
    pin: GPIO2
    id: buzz_out

# ============================================================================
# https://esphome.io/components/rtttl.html
rtttl:
  output: buzz_out
  id: buzzer

# ============================================================================

# https://esphome.io/components/sensor
sensor:
  - <<: !include common/sensor/uptime.config.yaml
  - <<: !include common/sensor/wifi_signal.config.yaml

  # https://esphome.io/components/sensor/adc.html#adc-esp8266-vcc
  - platform: adc
    pin: VCC
    name: "${friendly_name} VCC"

  # https://next.esphome.io/components/sensor/sdm_meter.html
  - platform: sdm_meter
    phase_a:
      voltage:
        name: "${friendly_name} Voltage"
        unit_of_measurement: V
        accuracy_decimals: 1
        icon: "mdi:gauge"
      current:
        name: "${friendly_name} Current"
        unit_of_measurement: A
        accuracy_decimals: 3
        icon: "mdi:gauge"
      active_power:
        name: "${friendly_name} Power"
        unit_of_measurement: W
        accuracy_decimals: 1
        icon: "mdi:gauge"
        id: power
      power_factor:
        name: "${friendly_name} Power Factor"
        accuracy_decimals: 2
        icon: "mdi:gauge"
    frequency:
      name: "${friendly_name} Frequency"
      unit_of_measurement: Hz
      accuracy_decimals: 1
      icon: "mdi:gauge"
    import_active_energy:
      name: "${friendly_name} Energy"
      unit_of_measurement: kWh
      accuracy_decimals: 1
      icon: "mdi:counter"

  # https://esphome.io/components/sensor/total_daily_energy.html
  - platform: total_daily_energy
    name: "${friendly_name} Daily Energy"
    icon: "mdi:counter"
    power_id: power
    filters:
      # Multiplication factor from W to kW is 0.001
      - multiply: 0.001
    unit_of_measurement: kWh

# ============================================================================

# https://esphome.io/components/status_led
status_led:
  pin:
    number: GPIO0
    inverted: yes

# ============================================================================

switch:
  - <<: !include common/switch/restart_switch.config.yaml

# ============================================================================
```


### Anything in the logs that might be useful for us?

```txt
$ esphome run powermeter111.yaml
INFO Reading configuration powermeter111.yaml...
INFO Detected timezone 'CET' with UTC offset 1 and daylight saving time from 27 March 02:00:00 to 30 October 03:00:00
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d powermeter111
Processing powermeter111 (board: esp01_1m; framework: arduino; platform: platformio/espressif8266 @ 2.6.2)
--------------------------------------------------------------------------------
Tool Manager: Installing platformio/tool-scons @ ~4.40200.0
Unpacking  [####################################]  100%
Tool Manager: tool-scons @ 4.40200.0 has been installed!
Tool Manager: Removing tool-scons @ 4.40100.2
Tool Manager: tool-scons @ 4.40100.2 has been removed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
PACKAGES:
 - framework-arduinoespressif8266 3.20704.0 (2.7.4)
 - tool-esptool 1.413.0 (4.13)
 - tool-esptoolpy 1.20800.0 (2.8.0)
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Library Manager: Installing Update
Library Manager: Already installed, built-in library
Dependency Graph
|-- <ESPAsyncTCP-esphome> 1.2.3
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncWebServer-esphome> 1.3.0
|   |-- <ESPAsyncTCP-esphome> 1.2.3
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266WiFi> 1.0
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
|-- <EspSoftwareSerial> 6.8.5
|-- <DNSServer> 1.1.1
|   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs\powermeter111\src\esphome\components\adc\adc_sensor.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\api_connection.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\api_frame_helper.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\api_pb2.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\api_pb2_service.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\api_server.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\list_entities.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\proto.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\subscribe_state.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\user_services.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\api\util.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\binary_sensor\automation.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\binary_sensor\binary_sensor.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\binary_sensor\filter.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\captive_portal\captive_portal.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\esp8266\core.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\esp8266\gpio.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\esp8266\preferences.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\esp8266_pwm\esp8266_pwm.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\homeassistant\time\homeassistant_time.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\json\json_util.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\logger\logger.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\mdns\mdns_component.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\mdns\mdns_esp32_arduino.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\mdns\mdns_esp8266.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\mdns\mdns_esp_idf.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\modbus\modbus.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\network\util.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\ota\ota_component.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\output\automation.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\output\float_output.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\restart\restart_switch.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\rtttl\rtttl.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\sdm_meter\sdm_meter.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\sensor\automation.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\sensor\filter.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\sensor\sensor.cpp.o
Compiling .pioenvs\powermeter111\src\esphome\components\socket\bsd_sockets_impl.cpp.o
src\esphome\components\rtttl\rtttl.cpp: In member function 'void esphome::rtttl::Rtttl::play(std::string)':
src\esphome\components\rtttl\rtttl.cpp:83:23: error: 'millis' was not declared in this scope
   last_note_ = millis();
                       ^
src\esphome\components\rtttl\rtttl.cpp: In member function 'virtual void esphome::rtttl::Rtttl::loop()':
src\esphome\components\rtttl\rtttl.cpp:88:37: error: 'millis' was not declared in this scope
   if (note_duration_ == 0 || millis() - last_note_ < note_duration_)
                                     ^
src\esphome\components\rtttl\rtttl.cpp:170:31: error: 'delay' was not declared in this scope
       delay(DOUBLE_NOTE_GAP_MS);
                               ^
src\esphome\components\rtttl\rtttl.cpp:183:23: error: 'millis' was not declared in this scope
   last_note_ = millis();
                       ^
Compiling .pioenvs\powermeter111\src\esphome\components\socket\lwip_raw_tcp_impl.cpp.o
*** [.pioenvs\powermeter111\src\esphome\components\rtttl\rtttl.cpp.o] Error 1
========================= [FAILED] Took 26.97 seconds =========================
```


### Additional information

_No response_

--- a/esphome/components/rtttl/rtttl.cpp
+++ b/esphome/components/rtttl/rtttl.cpp
@@ -1,4 +1,5 @@
 #include "rtttl.h"
+#include "esphome/core/hal.h"
 #include "esphome/core/log.h"
 
 namespace esphome {

